### PR TITLE
Shard stepping rework

### LINKED ISSUE #3058: Stepping on glass shards with no legs
[BUG]

**Describe the bug**
When both legs are removed, and wearing no shoes, you will still "step on glass shards" and be tripped.

**To Reproduce**
1. Have no legs
2. Be wearing no shoes
3. Be in wheelchair
4. Roll over glass shard
5. Be informed you have stepped on glass and fall over in wheelchair

**Expected behavior**
When having no legs and in a wheelchair, one should not be effected by glass shards on the floor as the wheelchair would roll over them.

### PR DESCRIPTION
**Pull request by BatElite**
[rework]
## About the PR 
(Jeez I'm productive lately)
This PR redoes the bit of code where you get hurt if you step on a glass shard. If you're lying on the floor now you get hurt regardless. (previously it'd be identical to standing) Stepping on one while you've got two robot legs or none (possible only in a chair) won't hurt you anymore.

Riding around over one in a chair with your organic legs still attached hurts you as normal though.

## Why's this needed? 
closes #3058

Shoes protecting you from glass shard while you're laying on the floor was silly, as is getting hurt from stepping on one with your robot feet. I demand being able to go barefeet with impunity if I'm willing to saw off my legs for it.

## Changelog

```
(u)BatElite:
(+)Reworked how stepping on shards works - wearing shoes doesn't save you if you crawl over them, but having robtic legs (or being in a chair with no legs) lets you go over them with no problem.
```

### CODE REVIEW

**Comment by ZeWaka** [comment]
definitely needs a changelog

**Comment by ZeWaka** [comment]
> Riding around over one in a chair with your organic legs still attached hurts you as normal though.

why? This seems improper.

File: code/obj/item/material.dm [hunk 1]
```diff
@@ -608,6 +609,16 @@
 			var/datum/material/M = getMaterial("plasmaglass")
 			src.setMaterial(M, appearance = 1, setname = 1)
 
+/obj/item/raw_material/shard/proc/step_on(mob/living/carbon/human/H as mob)
+	playsound(src.loc, src.sound_stepped, 50, 1)
+	var/obj/item/affecting = H.organs[pick("l_leg", "r_leg")]
+	H.changeStatus("weakened", 3 SECONDS)
+	H.force_laydown_standup()
+	var/shard_damage = force
+	affecting.take_damage(shard_damage, 0)
```
Comment (ZeWaka) [inline code/obj/item/material.dm:618]: pretty useless variable
```suggestion
	affecting.take_damage(force, 0)
```

**Comment by BatElite** [comment]
> why? This seems improper.

I've been trying to make it check if you're buckled in in general (now and before submitting a PR), but is not been working so far (always null) and I'm not sure why.

**Review by pali6 — APPROVED** [review]

**Pull request merged**